week2/homework/src/index.js

```diff
@@ -1,3 +1,134 @@
 'use strict';
+const fs = require('fs');
+const command = process.argv[2];
+const commandFirstItem = process.argv[3];
+const newData = process.argv[4];
 
-// TODO: Write the homework code in this file
+function help() {
+  fs.readFile('./help.txt', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      console.log(data);
+    }
+  });
+};
+
+function list() {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      if (Object.values(myToDoList).length === 0) {
+        console.log('You have not any plan in your To Do List');
+      }
+      else {
+        console.log(myToDoList);
+      }
+    }
+  });
+};
+
+function add(commandFirstItem) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      const index = Object.keys(myToDoList).length + 1;
+      myToDoList[index] = commandFirstItem;
+      fs.writeFile(
+        './toDo.json',
+        JSON.stringify(myToDoList),
+        function (error) {
+          if (error) {
+            console.error(error);
+          }
+        }
+      );
+    }
+  });
+}
+
+function remove(commandFirstItem) {
+  fs.readFile('./toDo.json', 'utf8', function (error, data) {
+    if (error) {
+      console.log(error);
+    }
+    else {
+      const myToDoList = JSON.parse(data);
+      if (commandFirstItem >= 1) {
+        delete myToDoList[parseInt(commandFirstItem)];
+        fs.writeFile(
+          './toDo.json',
+          JSON.stringify(myToDoList),
+          (error) => {
+            if (error) { console.log(error); }
+          });
+      }
+      else {
+        console.log('The parameter you entered as index is invalid');
+      }
+    }
+  });
+}
```

Comment:
To clean up a bit more this function (and the other similar functions) you can use something I use all the time, [early returns](http://blog.timoxley.com/post/47041269194/avoid-else-return-early).
In a function when you use `return` the function will immediately stop and exit with what you return.
Example refactoring the callback function inside `fs.readFile`:
```js
// in case of error, stop immediately
if(error) {
  return console.log(error);
}
if (commandFirstItem < 1) {
  return console.log('The parameter you entered as index is invalid');
}

const myToDoList = JSON.parse(data);
delete myToDoList[parseInt(commandFirstItem)];
fs.writeFile(
  './toDo.json',
  JSON.stringify(myToDoList),
  (error) => {
    if (error) { console.log(error); }
  });
```

This is an advanced trick, I don't want to confuse you.
The code is exactly the same as what you wrote, but it's cleaner and easier to follow because the main logic is not indented :)

Reply:
Yeah, I liked this view of early return. I have tried what I can do. I am curious about have I achieved it

Reply:
Yes I tried to edit it.